### DISCUSSION TITLE
Minifying breaks z-index rules

### DISCUSSION THREAD

**Discussion by emarthinsen** [tailwindlabs/tailwindcss#12738] [Help]
I have two tasks in my package.json, which are defined like this:

```
    "watch": "npx tailwindcss -i ./client/tailwind.css -o ./web/css/tailwind.css --watch",
    "tailwind": "npx tailwindcss -i ./client/tailwind.css -o ./web/css/tailwind.css --minify"
```

When I run `npm run watch` and inspect the output, I see that `z-50` has the CSS `z-index: 50;`.

When I run `npm run tailwind` and inspect the output, I see that `z-50` has the CSS `z-index: 2;`.

Does anyone know why this might be? I'm using a slightly older version of Tailwind (3.1.8).

**Comment by alex-krasikau** [comment]
Hello @emarthinsen!

Could you please share your Tailwind and PostCSS configurations for further investigation? Do you use any tools like cssnano or [postcss-zindex](https://www.npmjs.com/package/postcss-zindex) in your project?

This appears to be a similar issue that you can refer to for insights into the problem: [invalid z-index after postcss with minimize option](https://github.com/egoist/rollup-plugin-postcss/issues/73).

**Reply by emarthinsen** [reply]
Sure thing. Attached is our Tailwind config. I actually don't have a PostCSS config. Could that be the problem?

```
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    './templates/**/*.{html,twig,svg}',
    './client/elm/src/**/*.elm',
  ],

  theme: {
    colors: {
      brand: '#8C1F30',  // synonym for destructive-500
      transparent: 'transparent',
      white: '#FFFFFF',
      blue: '#1fb6ff',

      destructive: {
        50: '#FEF2F2',
        100: '#FEE2E2',
        200: '#FECACA',
        300: '#FCA5A5',
        400: '#F87171',
        500: '#8C1F30',
        600: '#DC2626',
        700: '#B91C1C',
        800: '#991B1B',
        900: '#7F1D1D',
      },
      neutral: {
        50: '#F2F7FF',
        100: '#D8E7FC',
        200: '#BFD1EB',
        300: '#9CB3D4',
        400: '#6B84A9',
        500: '#526785',
        600: '#405169',
        700: '#2E3B4E',
        800: '#253143',
        900: '#1B2636',
      },
      success: {
        50: '#F0FDF4',
        100: '#DCFCE7',
        200: '#BBF7D0',
        300: '#86EFAC',
        400: '#4ADE80',
        500: '#22C55E',
        600: '#16A34A',
        700: '#15803D',
        800: '#166534',
        900: '#166534',
      },
    },
    extend: {
      boxShadow: {
        std: '0px 4px 12px rgba(46, 59, 78, 0.6)',
      },
      spacing: {
        '128': '32rem',
        '144': '36rem',
      }
    },
  },

  plugins: [
    require('@tailwindcss/aspect-ratio'),
    require('@tailwindcss/forms'),
    require('@tailwindcss/typography'),
  ],
}
```

**Reply by alex-krasikau** [reply]
Do you have a bundler that compiles Elm and CSS? It would be helpful if you could share its configuration.

I will test version `3.1.8` later today. In the meantime, could you try upgrading to the latest version of Tailwind and see if the issue persists? Please ensure that you remove any build result files beforehand to prevent any interference with the build caused by items in the cache.

**Reply by emarthinsen** [reply]
@alex-krasikau We do use Gulp to compile Elm and some legacy SCSS, but Gulp doesn't get invoked when we compile our Tailwind styles. We haven't run any of our Gulp tasks in some time, since they really just support one page of the site that hasn't changed in a long time.
